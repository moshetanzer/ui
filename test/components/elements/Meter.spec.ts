import { describe, it, expect } from 'vitest'
import { UMeter } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Meter', () => {
  it.each([
    ['basic case', {}],
    ['with custom value', { props: { value: 50 } }],
    ['with custom min', { props: { min: 20 } }],
    ['with custom max', { props: { max: 200 } }],
    ['with indicator enabled', { props: { indicator: true } }],
    ['with label', { props: { label: 'meter label' } }],
    ['with icon', { slots: { label: () => 'label slot' }, props: { icon: 'i-heroicons-chart-bar' } }],
    ['with custom size', { props: { size: 'lg' } }],
    ['with custom color', { props: { color: 'red' } }],
    ['with label slot', { slots: { label: () => 'Label slot' } }],
    ['with indicator slot', { slots: { indicator: () => 'Indicator slot' }, props: { indicator: true } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UMeter.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UMeter)
    expect(html).toMatchSnapshot()
  })
})
