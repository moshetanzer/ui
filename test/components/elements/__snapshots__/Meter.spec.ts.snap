// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Meter > renders basic case correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with custom color correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-red-500 dark:text-red-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with custom max correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="200" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with custom min correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="20" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with custom size correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-3 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-3 [&::-moz-meter-bar]:h-3" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with custom value correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="50"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with icon correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>label slot
</div>"
`;

exports[`Meter > renders with indicator enabled correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <div class="min-w-fit transition-all" style="width: 0%;">
    <div class="text-gray-400 dark:text-gray-500 text-end text-sm">0% </div>
  </div><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with indicator slot correctly 1`] = `
"<div class="w-full flex flex-col gap-2">Indicator slot<meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <!--v-if-->
</div>"
`;

exports[`Meter > renders with label correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>
  <div class="flex gap-2 items-center truncate text-primary-500 dark:text-primary-400 text-sm">
    <!--v-if--> meter label
  </div>
</div>"
`;

exports[`Meter > renders with label slot correctly 1`] = `
"<div class="w-full flex flex-col gap-2">
  <!--v-if--><meter min="0" max="100" class="appearance-none block w-full bg-none overflow-y-hidden bg-gray-200 dark:bg-gray-700 rounded-full text-primary-500 dark:text-primary-400 h-2 [&::-webkit-meter-inner-element]:block [&::-webkit-meter-inner-element]:relative [&::-webkit-meter-inner-element]:border-none [&::-webkit-meter-inner-element]:bg-none [&::-webkit-meter-inner-element]:bg-transparent [&::-webkit-meter-bar]:border-none [&::-webkit-meter-bar]:bg-none [&::-webkit-meter-bar]:bg-transparent [&::-webkit-meter-optimum-value]:border-none [&::-webkit-meter-optimum-value]:bg-none [&::-webkit-meter-optimum-value]:bg-current [&::-moz-meter-bar]:border-none [&::-moz-meter-bar]:bg-none [&::-moz-meter-bar]:bg-current [&::-webkit-meter-optimum-value]:transition-all [&::-moz-meter-bar]:transition-all [&::-webkit-meter-optimum-value]:rounded-full [&::-moz-meter-bar]:rounded-full [&::-webkit-meter-optimum-value]:h-2 [&::-moz-meter-bar]:h-2" value="0"></meter>Label slot
</div>"
`;
